src/services/blockService.ts

```diff
@@ -0,0 +1,103 @@
+import { DataAccessLayer } from "@blockr/blockr-data-access";
+import { Block } from "@blockr/blockr-models";
+import { NextFunction } from "connect";
+import { Request, Response } from "express";
+import { inject, injectable } from "inversify";
+import { HttpException } from "../utils/exceptions/httpException";
+import logger from "../utils/logger";
+
+@injectable()
+export class BlockService {
+    protected dataAccessLayer: DataAccessLayer;
+
+    constructor(@inject(DataAccessLayer) dataAccessLayer: DataAccessLayer) {
+        this.dataAccessLayer = dataAccessLayer;
+    }
+
+    public async getBlocksAsync(request: Request, response: Response, next: NextFunction) {
+        try {
+            response.send(
+                request.query.number !== undefined ? await this.getBlockByNumberAsync(request.query.number)
+                : request.query.hash !== undefined ? await this.getBlocksByHashAsync(request.query.hash)
+                : request.query.dateperiod_start !== undefined && request.query.dateperiod_end !== undefined
+                    ? await this.getBlocksByDatePeriodAsync(request.query.dateperiod_start,
+                                                            request.query.dateperiod_end)
+                : request.query.parentHash !== undefined ? await this.getPreviousBlockAsync(request.query.parentHash)
+                : await this.getAllBlocksAsync());
+            next();
+        } catch (error) {
+            logger.error(error);
+            next(error);
+        }
+    }
+
+    private async getBlockByNumberAsync(numberBlock: number): Promise<Block> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                logger.info("Getting block by number.");
+
+                resolve(await this.dataAccessLayer.getBlockAsync(numberBlock));
+            } catch (error) {
+                logger.error(error.message);
+
+                reject(new HttpException(error.message, 404));
```

Comment:
The only thing happening in this `try` code block is a method calling the `Data Access Layer`. But when an error occurs it transforms the error into a HttpException. This is wrong as the exception can never be related to any HTTP protocol interaction.